### ISSUE TITLE
Can Karmada also be used to form a federation in two remote clusters formed by internal networks? How to achieve it?

### ISSUE THREAD

**Issue by Schwarao** [karmada-io/karmada#5125]
Can Karmada also be used to form a federation in two remote clusters formed by internal networks? How to achieve it?

**Comment by XiShanYongYe-Chang** [comment]
Is the network between the two clusters connected?

You need to select a cluster as the host cluster to deploy Karmada, and then run the karmadactl join command to add the cluster and another cluster to the deployed Karmada instance.

If you have additional clusters to deploy the host cluster would be a better choice.

**Comment by XiShanYongYe-Chang** [comment]
The network between the host cluster and member clusters must be connected so that resources can be distributed.

**Comment by Schwarao** [comment]
Is it possible to establish a VPN connection or use internal network penetration

**Comment by XiShanYongYe-Chang** [comment]
> Is it possible to establish a VPN connection or use internal network penetration

Karmada does not require a specific network connection mode, but only requires network connection.

**Comment by Schwarao** [comment]
But through this command: kubectl karmada token create -- print register command -- kubeconfig/etc/karmada/karmada apiserver.com
Is the generated token using an internal network connection? But the remote network is currently unable to access this IP. Is there any way to do so?

**Comment by Schwarao** [comment]
Or do you mean changing the IP on the token generated by this command to a public IP is enough to connect?
kubectl karmada register 10.0.0.58:32443 --token jcbljd.tfotd1ylodf7jtm6 --discovery-token-ca-cert-hash sha256:3bc5b22efbef8470f010f65b68388266e2b8f3ab62a0151900761cd12760d34f

**Comment by RainbowMango** [comment]
> Can Karmada also be used to form a federation in two remote clusters formed by internal networks?

Do you mean to let Karmada manage two clusters in an internal network? That means without a specific network connection(like VPN), Karmada can't take to them directly.

**Comment by Schwarao** [comment]
yes,The two clusters can only be accessed through a wide area network now, which means we need to use VPN to connect the two clusters, right

**Comment by XiShanYongYe-Chang** [comment]
I think it may be possible, can you help make a try.

By the way, is this your production environment usage scenario?

**Comment by Schwarao** [comment]
Okay, I'm currently testing

**Comment by oussexist** [comment]
Hello there , i actually almost asking the same question ( not exactly ) i have an on-prem cluster and a cloud cluster so , if i do the init on the on-prem cluster so it'll be the controlplane host cluster lets say i want to join members with create token command it'll give me the register command and use that command on the cloud cluster , it won't be able to register since the ip adresse is for a local network , althoguht the local cluster have network acces and can ping google.com just fine , what to do ?
I need to init on the cloud cluster and join the local one or what ? even tho if i  did init on the cloud cluster , the api server config ( karmada config file ) i should manually move it the local cluster ? and when joining give the path to the file i manually created  with the same content as the cloud karmada config inited ?

**Comment by RainbowMango** [comment]
@oussexist let's discuss your case at #5143

**Comment by XiShanYongYe-Chang** [comment]
> Okay, I'm currently testing

Hi @Schwarao did the test go well?

**Comment by Schwarao** [comment]
> > Okay, I'm currently testing
> 
> Hi @Schwarao did the test go well?

I couldn't do it, but I completed my requirement using karma init -- karma apiserver advertising address.
